src/QtUtils.h

```diff
@@ -35,6 +37,13 @@ namespace QtUtils {
 }; // namespace QtUtils
 }; // namespace Mumble
 
+template< typename T > using qt_unique_ptr = std::unique_ptr< T, decltype(&Mumble::QtUtils::deleteQObject) >;
+
+/// Creates a new unique_ptr with custom deleter for any given QObject*
+template< typename T > qt_unique_ptr< T > createQtUniquePtr(T *object) {
+	return qt_unique_ptr< T >{ object, Mumble::QtUtils::deleteQObject };
+}
```

Comment:
```suggestion
template< typename T, typename ... Args > qt_unique_ptr< T > make_qt_unique(Args &&... args) {
	static_assert(std::is_base_of<QObject, T>::value);

	return qt_unique_ptr< T >{ new T(args), Mumble::QtUtils::deleteQObject };
}
```

Reply:
I tried to implement this, but the build is failing when i call this for some template compiling reason... Can you please take a look at this and tell me what I am doing wrong when calling this function?

Reply:
Oh I might have missed to forward the arguments... but yeah, I'll have a look :+1:

Reply:
Ah okay, I see. You'll have to pass the type of the to-be-created object as an explicit template parameter (just as with `std::make_unique`):
```cpp
qt_unique_ptr< UserLocalNicknameDialog > userNickname = make_qt_unique<UserNicknameDialog>(sessionId, qmUserNicknameTracker);
```

Additionally, in the `make_qt_unique` function, the return statement should be replaced by
```cpp
return qt_unique_ptr< T >{ new T(std::forward(args)), Mumble::QtUtils::deleteQObject };
```
to enable perfect forwarding of the arguments.